src/Commands/QuitCommand.cpp

```diff
@@ -0,0 +1,28 @@
+#include "QuitCommand.hpp"
+#include "Channel.hpp"
+#include "Client.hpp"
+#include "Server.hpp"
+
+QuitCommand::QuitCommand(std::string source, std::vector<std::string> params,
+                         Client* client) {
```

Comment:
you need to add the protection to block the command if the user is not registered.
```c++
if (!client->isRegistered()) {
        client->sendMessage(Replies::ERR_NOTREGISTERED());
        throw ClientException();
    }
```